### ISSUE TITLE
Wrong type signature of subclass implementation of an abstract class's abstract method not caught

### ISSUE THREAD

**Issue by annesylvie** [microsoft/pyright#3996]
**Describe the bug**
Pyright does not catch when a subclass of an abstract base class implements an abstract method of the base class with the wrong type signature.

**To Reproduce**
Create an `ABC` class with an `abstractmethod` that has a specific signature, then implement that method with a different signature on another class deriving from the first one - see example below.

Tested with pyright version `1.1.273`.

**Expected behavior**
Mypy signals this as an issue:
`Return type "int" of "bar" incompatible with return type "str" in supertype "Foo"`
However, pyright does not raise any error.

**Screenshots or Code**
```python
from abc import ABC, abstractmethod


class Foo(ABC):
    @abstractmethod
    def bar(self) -> str:
        pass


class SpecificFoo(Foo):
    def bar(self) -> int: # I expect a type error here
        return 1
```

**Comment by erictraut** [comment]
This check is disabled by default in "basic" type checking. The diagnostic check is called `reportIncompatibleMethodOverride`, and you can [enable it](https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md) using the `python.analysis.diagnosticSeverityOverrides` setting in pylance. This rule is on by default in "strict" type checking mode, but you can enable it independently if you'd like. For a full list of diagnostic rules supported by pyright (the type checker that underlies pylance), refer to [this documentation](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#diagnostic-rule-defaults).
